pkg/operator/controller/dns/controller_test.go

```diff
@@ -15,17 +15,28 @@ import (
 )
 
 func TestPublishRecordToZones(t *testing.T) {
-	var tests = []struct {
-		name   string
-		zones  []configv1.DNSZone
-		expect []string
+	tests := []struct {
+		name         string
+		zones        []configv1.DNSZone
+		unManagedDNS bool
+		expect       []iov1.DNSZoneStatus
 	}{
 		{
 			name: "testing for a successful update of 1 record",
-			zones: []configv1.DNSZone{{
-				ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+			zones: []configv1.DNSZone{
+				{ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
 			},
-			expect: []string{string(operatorv1.ConditionFalse)},
+			expect: []iov1.DNSZoneStatus{{
+				DNSZone: configv1.DNSZone{
+					ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io",
+				},
+				Conditions: []iov1.DNSZoneCondition{
+					{
+						Type:   "Published",
```

Comment:
is this a reason these `"Published"` can't be `iov1.DNSRecordPublishedConditionType` instead of a hard-coded string?

Reply:
I have a couple of reasons, 
- Using the variable kind of doesn't ensure the value of the literal is exactly what we want. 
- And for literals that would almost never change, this would ensure the tests would break even if someone changed the value of the variable. 

Moreover, most of the other tests are also written similarly. (assuming its due to the same reasoning)

Reply:
Okay go it. Those are valid points.